hal/i2c_api.h

```diff
@@ -133,7 +133,8 @@ void i2c_reset(i2c_t *obj);
  *
  *  @param obj The I2C object
  *  @param last Acknoledge
- *  @return The read byte
+ *  @return The read byte if one byte is read, the number of bytes 
+ *        read if it is not one. 
```

Comment:
Hi - I'm not sure to fully understand the proposal. 
What could be the possible values of "the number of bytes read if it is not one".
If I return 0 : does it mean that I've read 1 byte whose value is 0, or that haven't read any byte ...